File: kdcreader_android/android/src/main/kotlin/koamtac/kdc/sdk/plugins/view/KViewFactory.kt
package koamtac.kdc.sdk.plugins.view

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class KViewFactory(private val messenger: BinaryMessenger) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    companion object {
        const val SW_DECODER_COMPONENT_VIEW_TYPE_ID = "SWComponentView"
    }

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val params = args as Map<*, *>?
        return KComponentView(context, viewId, messenger, params)
    }
}

File: kdcreader_ios/ios/Classes/view/KViewFactory.swift
//
//  KViewFactory.swift
//  device_info_plus
//
//  Created by koamtac on 4/3/24.
//

import Flutter

class KViewFactory: NSObject, FlutterPlatformViewFactory {
    static let SW_DECODER_COMPONENT_VIEW_TYPE_ID = "SWComponentView"
    
    private var messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return KComponentView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            binaryMessenger: messenger)
    }

    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
          return FlutterStandardMessageCodec.sharedInstance()
    }
}
